todo_app/data/trello_items.py

```diff
@@ -18,209 +29,203 @@ def __init__(
 
-# update a card on the board
-def update_card(cardId, listId):
-    searchUrl = baseUrl + f"cards/{cardId}"
-
-    addQuery = {
-        "key": os.getenv("TRELLO_KEY"),
-        "token": os.getenv("TRELLO_TOKEN"),
-        "idList": listId,
-    }
+    collectionCard= cardsCollection
+    collectionBoard = boardsCollection
+    if addedCollectionCard is not None:
+        collectionCard= addedCollectionCard
+        collectionBoard = addedCollectionBoard
     try:
-        response = requests.request("PUT", searchUrl, headers=headers, params=addQuery)
-        response.raise_for_status()
-        response = response.json()
-    except requests.exceptions.RequestException as exception:
-        print(f"An error occured: {exception}")
-        response = False
+        # First, find the document to get its _id
+        board = collectionBoard.find_one({"name": boardName}, {"_id": 1})
+        if board:
+            board_id_str = str(board['_id'])
+            # Attempt to delete the document by name
+            result = collectionBoard.delete_one({"_id": board['_id']})
+            
+            if result.deleted_count > 0:
+                print(f"Board '{boardName}' was deleted successfully.")
+
+                # Now, delete all related cards that have this board's _id in their 'boards' property
+                related_cards_result = collectionCard.delete_many({"boards": {"$in": [board_id_str]}})
+                print(f"Related cards deleted count: {related_cards_result.deleted_count}")
+
+                return True
+            else:
+                print(f"No board found with the name '{boardName}'.")
+                return False
+        else:
+            print(f"No board found with the name '{boardName}', nothing to delete.")
+            return False
+    except Exception as exception:
+        print(f"An error occurred: {exception}")
+        return False
+
+# Get all document from the boards collection
+def get_boards(addedCollection = None):
+
+    collection = boardsCollection
+    if addedCollection is not None:
+        collection = addedCollection
+    try:
+        boards = list(collection.find({}))  
+        return boards  
+    except Exception as exception:
+        print(f"An error occurred: {exception}")
+        return None
+    
+# update a card on the board
+def update_card(boardID, listId, addedCollection = None):
 
-    return response
+    collection = cardsCollection
+    if addedCollection is not None:
+        collection = addedCollection
 
+    try:
+        # The update_one method is used to update a single document.
+        result = collection.update_one(
+            {"_id": boardID},
+            {"$set": {"listId": listId}}
+        )
+        # Check if the document was found and updated
+        if result.matched_count > 0:
+            if result.modified_count > 0:
+                return {"status": "success", "message": "Document updated."}
```

Comment:
I'll admit, I'm eternally surprised that you've got so much code here that returns the various statuses but then we just ignore them all entirely (e.g. calls to `update_card` never look at the return value!) - for the sake of simplicity would we be better off removing these entirely? 
They may become more easily converted into log messages later though to be fair